modules/wifi_app_util/src/wifi_app_util.c

```diff
@@ -0,0 +1,93 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* SPDX-FileCopyrightText: 2025 Silicon Laboratories Inc. */
+
+#include <zephyr/net/wifi.h>
+#include <stdint.h>
+
+uint8_t wifi_app_map_security_to_siconnect(enum wifi_security_type security)
+{
+	uint8_t code = 0xFF; /* Unknown */
+
+	switch (security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		code = 0; /* OPEN */
+		break;
+	case WIFI_SECURITY_TYPE_WPA_PSK:
+		code = 1; /* WPA */
```

Comment:
It'd be be better to add defines for these magic values, if there's no header file where they're already defined. Once you have those you wont need the separate code comments anymore.